src/kaleidoscope/plugin/IdleLEDs.cpp

```diff
@@ -56,7 +59,47 @@ EventHandlerResult IdleLEDs::onKeyswitchEvent(Key &mapped_key, KeyAddr key_addr,
   return EventHandlerResult::OK;
 }
 
+uint16_t PersistentIdleLEDs::settings_base_;
+
+EventHandlerResult PersistentIdleLEDs::onSetup() {
+  settings_base_ = ::EEPROMSettings.requestSlice(sizeof(uint16_t));
+
+  // If idleTime is max, assume that EEPROM is uninitialized, and store the
+  // defaults.
+  uint16_t idle_time;
+  KeyboardHardware.storage().get(settings_base_, idle_time);
+  if (idle_time == 0xffff) {
+    KeyboardHardware.storage().put(settings_base_, idle_time);
+  }
+  setIdleTimeoutSeconds(idle_time);
+
+  return EventHandlerResult::OK;
+}
+
+void PersistentIdleLEDs::setIdleTimeoutSeconds(uint32_t new_limit) {
+  IdleLEDs::setIdleTimeoutSeconds(new_limit);
+  KeyboardHardware.storage().put(settings_base_, (uint16_t)new_limit);
+}
+
+EventHandlerResult PersistentIdleLEDs::onFocusEvent(const char *command) {
+  if (::Focus.handleHelp(command, PSTR("idleLeds.idleTimeLimit")))
+    return EventHandlerResult::OK;
+
+  if (strcmp_P(command, PSTR("idleLeds.idleTimeLimit")) != 0)
```

Comment:
FWIW, the way this is implemented, the string "idleLeds.idleTimeLimit" is stored twice in PROGMEM.

Reply:
Right. I'll clean that up when I get back to this PR.